indico/htdocs/js/indico/modules/registration/registration.js

```diff
@@ -108,11 +115,39 @@
         $('#select-none').on('click', function() {
             $('table.i-table input.select-row').prop('checked', false).trigger('change');
         });
+
+        $('.js-dialog-send-email').ajaxDialog({
+            getExtraData: function() {
+                var ids = getSelectedRows();
+                return {registration_ids: ids};
+            }
+        });
+
+        $('.registrations').on('indico:confirmed', '.js-delete-registrations', function(evt) {
+            evt.preventDefault();
+            var $this = $(this);
+            var selectedRows = getSelectedRows();
+            $.ajax({
+                url: $this.data('href'),
+                method: $this.data('method'),
+                data: {registration_ids: selectedRows},
+                traditional: true,
+                complete: IndicoUI.Dialogs.Util.progress(),
+                error: handleAjaxError,
+                success: function() {
+                    $('.registrations input:checkbox:checked').each(function() {
```

Comment:
That won't work when the request is slow and someone changes the checkboxes in the meantime. Why not use the IDs instead (simply give each row `id="registration-XXX"`)? Or just be lazy and re-render the whole list.